include/seastar/core/seastar.hh

```diff
@@ -384,4 +385,26 @@ future<uint64_t> fs_avail(std::string_view name) noexcept;
 future<uint64_t> fs_free(std::string_view name) noexcept;
 /// @}
 
+namespace experimental {
+/// \defgroup subprocess-module Subprocess Management
+///
+/// Seastar provides a simple subprocess management API to
+/// spawn and interact with a subprocess.
+///
+/// \addtogroup subprocess-module
+/// @{
+
+/// Spawn a subprocess
+///
+/// \param pathname the path to the executable
+/// \param params optional parameters for spawning the subprocess
+///
+/// \return a process representing the spawned subprocess
+/// \note
+/// the subprocess is spawned with \c posix_spawn() system call, so the
+/// pathname should be relative or absolute path of the executable.
+future<process> spawn_process(const std::filesystem::path& pathname,
+                              spawn_parameters params = {});
```

Comment:
spawn_parameters isn't defined yet.

Reply:
it's defined by the included `seastar/util/process.hh`. the purpose of including it is to make `seastar.hh` self-contained. but yes, this is inconsistent with how we, for instance, implement the `connected_socket` related API. i replaced `#include <seastar/util/process.hh>` with forward declarations. but i still think it'd be better to be self-contained, so user does not need to include other headers when using API's declared by `seastar.h`. if this is desirable, i will create another change to address this.

Reply:
Missed it. It's more accurate to have spawn_parameters.hh so that seastar.hh is kept thin, but not worth it for now.

Reply:
but for accessing `process` one still needs to include `seastar/util/process.hh`. that's exactly what https://github.com/scylladb/seastar/blob/e9511e1b9e9596ea167e9a4001104c83c2c5a032/tests/unit/spawn_test.cc#L22-L25 is doing.